File: Loenn/entities/boost_bumper.lua
local utils = require "utils"

local boostBumper = {}

boostBumper.name = "vitellary/boostbumper"
boostBumper.depth = 0

boostBumper.nodeLimits = {0, 1}
boostBumper.nodeLineRenderType = "line"

boostBumper.placements = {
    name = "boost_bumper"
}

boostBumper.texture = "objects/boostBumper/booster00"

function boostBumper.selection(room, entity)
    local x, y = entity.x or 0, entity.y or 0
    return utils.rectangle(x - 9, y - 9, 18, 18)
end

return boostBumper

File: Ahorn/entities/boostBumper.jl
module BoostBumper

using ..Ahorn, Maple

@mapdef Entity "vitellary/boostbumper" Booster(x::Integer, y::Integer)

const placements = Ahorn.PlacementDict(
    "Bumper Booster (Crystalline)" => Ahorn.EntityPlacement(
        Booster
    )
)

function Ahorn.selection(entity::Booster)
    x, y = Ahorn.position(entity)

    return Ahorn.Rectangle(x-9, y-9, 18, 18)
end

Ahorn.render(ctx::Ahorn.Cairo.CairoContext, entity::Booster) = Ahorn.drawSprite(ctx, "objects/boostBumper/booster00.png", 0, 0, jx=0.5, jy=0.5)

end